pint/compat.py

```diff
@@ -392,3 +193,216 @@ def zero_or_nan(obj: Any, check_all: bool) -> bool | Iterable[bool]:
     if check_all and is_duck_array_type(type(out)):
         return out.all()
     return out
+
+
+# TODO: remove this warning after v0.10
+class BehaviorChangeWarning(UserWarning):
```

Comment:
This is dead code I think we can safely remove it

Reply:
I think so too. The same hapens with certain imports in compat (Self, Never, etc) but I would rather keep it as is until downstream libraries are aware. Again my suggestion is that we use this commit to migrate as is